lib/pure/json.nim

```diff
@@ -1010,8 +1010,14 @@ when defined(nimFixedForwardGeneric):
       dst = T(jsonNode.num)
 
   proc initFromJson[T: enum](dst: var T; jsonNode: JsonNode; jsonPath: var string) =
-    verifyJsonKind(jsonNode, {JString}, jsonPath)
-    dst = parseEnum[T](jsonNode.getStr)
+    verifyJsonKind(jsonNode, {JInt, JString}, jsonPath)
+    if jsonNode.kind == JInt:
+      let i = jsonNode.getBiggestInt
+      if (i <= low(T).int or i >= high(T).int) or $i.T == $i & " (invalid data!)":
```

Comment:
i would like to know if there is a better way to do the 2nd check here

Reply:
Wait what? You check against "(invalid data!)"? That can't be right. For the other checks use ``i <= low(T).BiggestInt`` etc.

Reply:
I'm trying to catch the case when an enum has holes within them, ex a = 1, b = 3, c = 4...

Reply:
That's beyond what we should support. Enum with holes are a deadend.

Reply:
`Enum with holes are a deadend` i don't understand what you mean by this...